### PR TITLE
Device migration accross backends or brokers

### PR DESCRIPTION
**Pull request by btry**
Create policies to allow administrators to pass agents from a backend to an other, and / or change MQTT settings.

### CODE REVIEW

**Comment by btry** [comment]
Admins need to monitor device migration. This PR will give a better service with feature/activity

**Comment by ajsb85** [comment]
This branch has conflicts that must be resolved

**Comment by btry** [comment]
This branch migh be obsoleted because I don't find a satisfying solution to migrate accross backends.

Maybe I should retrieved from this one the code to change from a broker to an other, and work later how to change migrate to an other backend.

Just keep this PR open to not forget it until I build the new branch

File: install/upgrade/update_to_dev.php [hunk 1]
```diff
@@ -50,13 +50,11 @@ function plugin_flyvemdm_update_to_dev(Migration $migration) {
    ]);
    $profileRight->updateProfileRights($profiles_id, $newRights);
 
-   // remove download base URL setting
-   Config::deleteConfigurationValues('flyvemdm', ['deploy_base_url']);
-
```
Comment (btry) [inline install/upgrade/update_to_dev.php]: moved at the end of all updates related to configuraion

File: install/upgrade/update_to_dev.php [hunk 2]
```diff
@@ -202,12 +217,4 @@ function plugin_flyvemdm_update_to_dev(Migration $migration) {
...
    $table = 'glpi_plugin_flyvemdm_mqttupdatequeues';
    $migration->dropKey($table, 'status');
-
-   $policyCategory = new PluginFlyvemdmPolicyCategory();
-   $policyCategory->import([
-      'completename'    => 'Security > USB'
-   ]);
-   $policyCategory->import([
-      'completename'    => 'Phone'
-   ]);
```
Comment (btry) [inline install/upgrade/update_to_dev.php]: creation of new policy categories is automatic from the list pf policies now. These lines are useless

**Comment by btry** [comment]
More configuration options
- backend and client may use TLS or not independently (on different ports)
image

**Comment by ajsb85** [comment]
Very good

**Pull request merged**